firmware/library/L1_Drivers/dac_test.cpp

```diff
@@ -0,0 +1,89 @@
+    PinConfigure::pin_map =
+        reinterpret_cast<PinConfigure::PinMap_t *>(&local_iocon);
+    PinConfigure test_pin(0, 26);
+    Mock<PinConfigureInterface> mock_dac_pin;
+    Fake(Method(mock_dac_pin, SetPinFunction),
+         Method(mock_dac_pin, EnableDac),
+         Method(mock_dac_pin, SetAsAnalogMode),
+         Method(mock_dac_pin, SetPinMode));
+    PinConfigureInterface & dac = mock_dac_pin.get();
+    Dac test_subject00(&dac);
+    Dac test_subject01;
+    SECTION("Initialize Dac")
+    {
+        // Source: "UM10562 LPC408x/407x User manual" table 686 page 814
+        constexpr uint8_t kBiasMask = 0b0;
+        local_dac_port.CR           = kBiasMask;
+        // Mocked out Initialize for the Verify Methods
+        test_subject00.Initialize();
+        // Initialize that can properly check the registers if they
+        // recieved the correct values.
+        test_subject01.Initialize();
+        // Check Pin Mode DAC_OUT
+        Verify(Method(mock_dac_pin, SetPinFunction).Using(Dac::kDacMode),
+        Method(mock_dac_pin, EnableDac).Using(true),
+        Method(mock_dac_pin, SetAsAnalogMode).Using(true),
+        Method(mock_dac_pin,
+               SetPinMode).Using(PinConfigureInterface::kInactive));
+    }
+    SECTION("Write Dac")
+    {
+        // Source: "UM10562 LPC408x/407x User manual" table 686 page 814
+        constexpr uint16_t kBiasMask   = 0b0000000000;
+        local_dac_port.CR           = kBiasMask;
+        // Test the WriteDac function by checking if the CR
+        // register receives the correctValue when the
+        // function has its value input.
+        constexpr uint8_t kDacWrite0 = 10;
+        constexpr uint8_t kDacWrite1 = 100;
+        test_subject00.WriteDac(10);
+        CHECK((local_dac_port.CR >> Dac::kDacOutReg) == kDacWrite0);
+        test_subject00.WriteDac(100);
+        CHECK((local_dac_port.CR >> Dac::kDacOutReg) == kDacWrite1);
+    }
+    SECTION("SetVoltage")
+    {
+        // Source: "UM10562 LPC408x/407x User manual" table 686 page 814
+        constexpr uint16_t kBiasMask   = 0b0000000000;
+        constexpr float kVoltageInput0 = 3;
+        constexpr float kDacVoltage0 = (kVoltageInput0 * 1024.0f) / 3.3f;
+        constexpr int kConversion0 = static_cast<int>(kDacVoltage0);
+        local_dac_port.CR           = kBiasMask;
+        test_subject00.SetVoltage(kVoltageInput0);
+        CHECK((local_dac_port.CR >> Dac::kDacOutReg) == kConversion0);
+    }
+    SECTION("SetBias")
+    {
+        // Source: "UM10562 LPC408x/407x User manual" table 686 page 814
+        constexpr uint8_t kBiasMask = 0b0;
+        constexpr uint8_t kBias0    = 0b0;
+        constexpr uint8_t kBias1    = 0b1;
+        local_dac_port.CR           = kBiasMask;
+        // Test Setting Bias to 0 and checking if the register is correct
+        test_subject00.SetBias(Dac::BiasLevel::kBiasHigh);
+        CHECK((local_dac_port.CR >> Dac::kBiasReg) == kBias0);
```

Comment:
This shouldn't be a shift. It should be an AND function. This goes for all tests. See the other tests by side effect examples like pin configure and pwm.

Reply:
Oh your right, I changed it to just be a delay so it shouldn't be long